src/reanimated2/commonTypes.ts

```diff
@@ -131,7 +131,11 @@ export function isWorkletFunction<
   'worklet';
   // Since host objects always return true for `in` operator, we have to use dot notation to check if the property exists.
   // See https://github.com/facebook/hermes/blob/340726ef8cf666a7cce75bc60b02fa56b3e54560/lib/VM/JSObject.cpp#L1276.
-  return !!(value as Record<string, unknown>).__workletHash;
+  return (
+    typeof value === 'function' &&
+    '__workletHash' in value &&
+    !!value.__workletHash
+  );
```

Comment:
The reason why we check `!!value.__workletHash` instead of `'__workletHash' in value` is that `in` operator always returns `true` for `jsi::HostObject`. We actually take advantage of that in `isHostObject` function:

https://github.com/software-mansion/react-native-reanimated/blob/4a15cf467b709a9365ae57a8eb7ba28397984a29/src/reanimated2/shareables.ts#L25-L32

Since we first check `typeof value === 'function'`, I guess it's safe to assume that once it passes, `value` is not a `jsi::HostObject`, however it could be a `jsi::HostFunction` and I'm not sure how `in` operator behaves in this case.

If possible, I would avoid changing that and I recommended reverting to the previous version (just with correct order of checks):

```suggestion
  return (
    typeof value === 'function' &&
    !!(value as Record<string, unknown>).__workletHash;
  );
```